packages/MapFactory/src/DTK_C_API_Map.hpp

```diff
@@ -83,10 +85,45 @@ struct DTK_MapImpl : public DTK_Map
             target_nodes.extent( 1 ) );
         Kokkos::deep_copy( target_nodes_copy, target_nodes );
 
-        // Create a nearest neighbor operator.
-        _map = std::unique_ptr<NearestNeighborOperator<map_device_type>>(
-            new NearestNeighborOperator<map_device_type>(
-                teuchos_comm, source_nodes_copy, target_nodes_copy ) );
+        auto const which_map =
+            ptree.get<std::string>( "Map Type", "Undefined" );
+        if ( which_map == "Undefined" )
+            throw DataTransferKitException(
+                R"(Field "Map Type" is not defined in options string argument for map creation)" );
+        else if ( which_map == "Nearest Neighbor" || which_map == "NN" )
+            _map = std::unique_ptr<NearestNeighborOperator<map_device_type>>(
+                new NearestNeighborOperator<map_device_type>(
+                    teuchos_comm, source_nodes_copy, target_nodes_copy ) );
+        else if ( which_map == "Moving Least Squares" || which_map == "MLS" )
+        {
+            // NOTE if field "Order" is misspelled (for instance first letter
```

Comment:
Should we maybe have a warning that no options were detected?

Reply:
@sslattery This happens because @dalg24 put a default value. If we don't have a default value, we don't have this problem

@aprokop Looks like someone as too much time on his hands :smiley:

Reply:
@Rombur I did it for MueLu, and people liked it. One does need to have the list of all options in one place, though. If #92 is ever done, this should be easy to implement.

Reply:
> Should we maybe have a warning that no options were detected?

I'd rather have no default if you think that's an issue